src/modules/SdnDiag.Health.psm1

```diff
@@ -2760,6 +2761,37 @@ function Test-SdnConfigurationState {
     }
 }
 
+function Test-SdnNetworkControllerNodeRestInterface {
+    <#
+        .SYNOPSIS
+            Validates that a Network Adapter on the Network Controller node exists that matches the RestInterface name.
+    #>
+
+    [CmdletBinding()]
+    param()
+
+    Confirm-IsNetworkController
+    if ($Global:SdnDiagnostics.EnvironmentInfo.ClusterConfigType -ieq 'FailoverCluster') {
+        throw New-Object System.NotSupportedException("This function is not applicable to Failover Cluster Network Controller.")
+    }
+
+    $sdnHealthTest = New-SdnHealthTest
+    try {
+        $node = Get-SdnNetworkControllerNode -Name $env:COMPUTERNAME -ErrorAction Stop
+        $netAdapter = Get-NetAdapter -Name $node.RestInterface -ErrorAction Ignore
```

Comment:
How does this work ? isnt it suppose to be executed on the network controller itself ?

Reply:
Get-SdnNetworkControllerNode is a wrapper around Get-NetworkController node that has fault tolerance if the SF cluster is down. -Name is used to return back the node that matches the local node, rather than all nodes in the cluster.